src/services/stellar.ts

```diff
@@ -58,6 +70,51 @@ export const stellarSdkServer = (networkUrl: string): Horizon.Server =>
     allowHttp: getIsAllowHttp(networkUrl),
   });
 
+/**
+ * Creates a Soroban RPC server instance for the given network
+ *
+ * @param network The network to get the Soroban RPC server for
+ * @returns A Soroban RPC server instance or null if there's an error
+ */
+export const getSorobanRpcServer = (network: NETWORKS) => {
+  try {
+    // Get the soroban RPC URL from the constants
+    const sorobanRpcUrl = SOROBAN_RPC_URLS[network];
+
+    if (!sorobanRpcUrl) {
+      logger.error(
+        "StellarService",
+        "No Soroban RPC URL available for network",
+        { network },
+      );
+      return null;
+    }
+
+    // Use a try-catch specifically for the Server instantiation
+    try {
+      return new rpc.Server(sorobanRpcUrl, {
+        allowHttp: getIsAllowHttp(sorobanRpcUrl),
+      });
+    } catch (serverError) {
+      logger.warn(
+        "StellarService",
+        "Failed to instantiate Soroban RPC Server",
+        {
+          error: String(serverError),
+        },
+      );
+      return null;
+    }
+  } catch (error) {
+    logger.error(
+      "StellarService",
+      "Failed to create Soroban RPC server",
+      String(error),
+    );
+    return null;
```

Comment:
do we need this catch here? I don't see a path where we could get here since the `new rpc.Server()` call has its own `try catch` block

Reply:
I think it wouldn't hurt to leave it here as boilerplate for future changes, but mentioning just in case it was an unintended leftover